tests/python/pants_test/backend/jvm/tasks/jvm_compile/zinc/zinc_compile_integration_base.py

```diff
@@ -343,6 +343,51 @@ def test_zinc_logs_warnings_properly(self):
         for expected in expected_strings:
             self.assertIn(expected, pants_run.stdout_data)
 
+    def _test_zinc_reports_diagnostic_counts(self, reporting):
+        with self.temporary_workdir() as workdir:
+            target = "testprojects/src/scala/org/pantsbuild/testproject/compilation_warnings/unused_import_warning:unused_import"
+            with self.temporary_cachedir() as cachedir:
+                args = ['--compile-rsc-args=+["-S-Ywarn-unused:_"]'] + (
+                    ["--compile-rsc-report-diagnostic-counts"] if reporting else []
+                )
+                pants_run = self.run_test_compile(workdir, cachedir, target, extra_args=args,)
+                self.assert_success(pants_run)
+
+            expected_strings = [
+                "Reporting number of diagnostics for: ScalaLibrary(testprojects/src/scala/org/pantsbuild/testproject/compilation_warnings/unused_import_warning:unused_import)",
```

Comment:
```suggestion
                f"Reporting number of diagnostics for: {target}",
```

Reply:
Not exactly. It needs to be `ScalaLibrary({target})`, but good point

Reply:
Ah OK. Read the comments in the wrong order :)